tdb/parse.py

```diff
@@ -42,32 +43,39 @@ def read_flat(self, path, fstem, **kwargs):
         else:
             with open(file) as infile:
                 table_reader = csv.reader(infile, delimiter="\t")
-                header = {
-                    0: 'virus_strain',
-                    1: 'serum_strain',
-                    2: 'ferret_id',
-                    3: 'source',
-                    4: 'titer'
-                }
+                header = table_reader.next()
+                headers = {}
+                for i in range(len(header)):
+                    headers[i] = header[i]
+                # header = {
+                #     0: 'virus_strain',
+                #     1: 'serum_strain',
+                #     2: 'ferret_id',
+                #     3: 'source',
+                #     4: 'titer'
+                # }
                 for row in table_reader:
-                    m = {key: row[ii] if ii < len(row) else "" for ii, key in header.items()}
-                    if re.search(r'[Ee][Gg][Gg]', m['ferret_id']):  # TODO FIX THIS FOR LATER IMPORTS
+                    m = {key: row[ii] if ii < len(row) else "" for ii, key in headers.items()}
+                    if 'ferret_id' in m.keys():
```

Comment:
@trvrb This one should actually be `ferret_id`, this is looking for names that were scraped from the table, and in the next lines it changed the `ferret_id` field to be `serum_id`.